File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/movie/model/MovieReleaseDateRegion.java
package dev.reuss.tmdb.domain.movie.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Release dates for a specific country.
 *
 * @param iso31661     ISO 3166-1 country code
 * @param releaseDates release dates
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record MovieReleaseDateRegion(
        @JsonProperty("iso_3166_1")
        String iso31661,

        @JsonProperty("release_dates")
        List<MovieReleaseDate> releaseDates
) implements TmdbModel {

    public MovieReleaseDateRegion {
        releaseDates = TmdbCollections.list(releaseDates);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/movie/model/MovieReleaseDateRegion.kt
package dev.reuss.tmdb.domain.movie.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Release dates for a specific country.
 *
 * @property iso31661 ISO 3166-1 country code
 * @property releaseDates release dates
 */
@JvmRecord
data class MovieReleaseDateRegion(
    @all:JsonProperty("iso_3166_1")
    val iso31661: String?,

    @all:JsonProperty("release_dates")
    val releaseDates: List<MovieReleaseDate> = emptyList()
) : TmdbModel